src/env.zig

```diff
@@ -92,3 +121,36 @@ else if (isWasm)
     OperatingSystem.wasm
 else
     @compileError("Please add your OS to the OperatingSystem enum");
+
+pub const Archictecture = enum {
+    x64,
+    arm64,
+    wasm,
+
+    pub fn npmName(this: Archictecture) []const u8 {
+        return switch (this) {
+            .x64 => "x64",
+            .arm64 => "aarch64",
+            .wasm => "wasm",
+        };
+    }
+
+    pub const names = @import("root").bun.ComptimeStringMap(
+        Archictecture,
+        &.{
+            .{ "x86_64", Archictecture.x64 },
+            .{ "x64", Archictecture.x64 },
+            .{ "amd64", Archictecture.x64 },
+            .{ "aarch64", Archictecture.arm64 },
+            .{ "arm64", Archictecture.arm64 },
```

Comment:
should the following be added

- `arm`
- `x86`
- `intel`

Reply:
no, we just fail when invalid now